### PR TITLE
General fixes and coupled compile on travis

### PR DESCRIPTION
**Pull request by ghost**
I fixed a few minor bugs related to the compile flags for the coupled build. The change to the preprocessing directive was scripted via sed for travis. Also, I changed the travis configuration to use MOM6-examples and do the coupled build.

### CODE REVIEW

**Comment by ghost** [comment]
Also, I added a travis badge to the README. Very important, obviously.

File: .travis.yml [hunk 1]
```diff
@@ -8,11 +8,12 @@ addons:
       - cmake
       - gfortran
       - libnetcdf-dev
+      - libnetcdff-dev
```
Comment (angus-g) [inline .travis.yml]: Did you mean to re-introduce this package here? (Since there was a commit explicitly removing it)

Comment (ghost) [reply .travis.yml]: I added this package by mistake, and afterwards removed it. On Debian, the fortran headers are located in it. On Ubuntu everything is included in `libnetcdf-dev`.

**Comment by angus-g** [comment]
Looks good to me, thanks!

**Pull request merged**